api/src/main/java/org/pmiops/workbench/api/ConceptSetsController.java

```diff
@@ -358,6 +360,63 @@ private void addConceptsToSet(
     return ResponseEntity.ok(toClientConceptSet(dbConceptSet));
   }
 
+  @Override
+  public ResponseEntity<ConceptSet> copyConceptSet(
```

Comment:
Just thinking out loud, but it might be possible to have common code for copying objects (of all types) and have that call object-type-specific code. That might let you consolidate implementation and testing for several different copy APIs on the different controllers. I.e. if they all use the same namespace rules and name conventions, then just having an object type parameter might let you share a lot of code.

Reply:
That sounds reasonable to me. Might be a good cleanup down the line. I think there's some different requirements that could require design though. (If I copy a data set, does it copy all underlying resources? How about if I copy a cohort review? I think there's some product stuff to figure out, but I worry about fitting too tightly to our model.)